shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/context/ProxyContext.java

```diff
@@ -115,7 +116,13 @@ public ShardingSphereMetaData getMetaData(final String schemaName) {
         if (schemaNames.isEmpty()) {
             return Optional.empty();
         }
-        Map<String, DataSource> dataSources = getMetaData(schemaNames.get(0)).getResource().getDataSources();
-        return dataSources.values().stream().findFirst();
+        for (String schemaName : schemaNames) {
+            ShardingSphereResource shardingSphereResource = getMetaData(schemaName).getResource();
+            DatabaseType databaseType = shardingSphereResource.getDatabaseType();
+            if ("MySQL".equals(databaseType.getName())) {
+                return shardingSphereResource.getDataSources().values().stream().findFirst();
+            }
```

Comment:
This is the public module, we cannot consider about MySQL only.
What is the behaviour of PostgreSQL?

Reply:
This method is only used by mysql, PostgreSQLdoes not set the version information.

Reply:
The method(getDataSourceSample) name can be changed。

Reply:
The ProxyContext is public class which is in shardingsphere-proxy-backend module.
The database dialect should in their own module, it is better to keep public module independent, and let them do not know database dialect details.

Reply:
This method(getDataSourceSample) is moved from ProxyContext  to AbstractBootstrapInitializer ，and changed to private
method，only used by setDatabaseServerInfo。

Reply:
please revivew again.